_includes/pbc/all/install-features/202404.0/install-dynamic-multistore.md

```diff
@@ -0,0 +1,1399 @@
+     */
+    protected function getStoreViewExpanderPlugins(): array
+    {
+        return [
+            new DefaultLocaleStoreViewExpanderPlugin(),
+            new AssignedLocalesStoreViewExpanderPlugin(), 
+            new AssignedCountriesStoreViewExpanderPlugin(),
+        ];
+    }
+
+    /**
+     * @return array<\Spryker\Zed\StoreGuiExtension\Dependency\Plugin\StoreTableExpanderPluginInterface>
+     */
+    protected function getStoreTableExpanderPlugins(): array
+    {
+        return [
+            new LocaleStoreTableExpanderPlugin(), 
+            new CountryStoreTableExpanderPlugin(),
+        ];
+    }
+}
+
+```
+
+{% info_block warningBox "Verification" %}
+
+Steps to verify:
+- Make sure that you can see the locale selection fields on the Store form.
+- Make sure that you can see the country selection fields on the Store form.
+- Make sure that you can see the rendered locale tabs and tables on the Store form.
+- Make sure that you can see the rendered country tabs and tables on the Store form.
+- Make sure that you can see the Locales tab on the Store form.
+- Make sure that you can see the Countries tab on the Store form.
+- Make sure that you can see the default locale ISO code on the Store view page.
+- Make sure that you can see the table with assigned locales on the Store view page.
+- Make sure that you can see the table with assigned countries on the Store view page.
+- Make sure that you can see the locale codes in the store table.   
+- Make sure that you can see the countries in the store table.
+
+{% endinfo_block %}
+
+## Install feature frontend
+
+### Prerequisites
+
+Install the required features:
+
+| NAME | VERSION |
+| --- | --- |
+|Spryker Core  | {{page.version}} |
+
+### 1) Install the required modules
+
+Install the required modules using Composer:
+
+```bash
+composer require "spryker-feature/spryker-core":"{{page.version}}" --update-with-dependencies
+```
+
+{% info_block warningBox "Verification" %}
+
+Make sure the following modules have been installed:
+
+| MODULE | EXPECTED DIRECTORY |
+| --- | --- |
+| StoreWidget | vendor/spryker-shop/store-widget | 
+
+{% endinfo_block %}
+
+
+### 2) Add translations
+
+Append glossary according to your configuration:
+
+**data/import/common/common/glossary.csv**
+
+
+```csv
+store_widget.switcher.store,Store:,en_US
+store_widget.switcher.store,Shop:,de_DE
+```
+
+Import data:
+
+```bash
+console data:import glossary
+```
+
+{% info_block warningBox "Verification" %}
+
+Make sure that, in the database, the configured data are added to the `spy_glossary` table.
+
+{% endinfo_block %}
+
+
+### 2) Set up configuration
+
+Add the following configuration to your project:
+
+| CONFIGURATION | SPECIFICATION | NAMESPACE |
+| --- | --- | --- |
+| RouterConfig::getAllowedLanguages() |  Returns a list of supported languages for Route manipulation. Will be used to strip of language information from a route before a route is matched. | Spryker\Yves\Router |
```

Comment:
btw we miss this also in the code, but this method should also be replaced with smth similar `\Spryker\Yves\Router\RouterConfig::getAllowedStores()`

Reply:
It's not clear how exactly to redeclare it. StoreClient does not have a similar method. Should we use `..->getCurrentStore()->getStoreName()` instead?

Reply:
I found this one \Spryker\Client\StoreStorage\StoreStorageClientInterface::getStoreNames